src/daft-local-execution/src/intermediate_ops/intermediate_op.rs

```diff
@@ -75,11 +82,20 @@ impl IntermediateNode {
         rt_context: Arc<RuntimeStatsContext>,
     ) -> DaftResult<()> {
         let span = info_span!("IntermediateOp::execute");
-        let mut state = op.make_state();
+        let compute_runtime = crate::get_compute_runtime()?;
+        let state = Arc::new(Mutex::new(op.make_state()));
         while let Some((idx, morsel)) = receiver.recv().await {
             loop {
-                let result =
-                    rt_context.in_span(&span, || op.execute(idx, &morsel, state.as_mut()))?;
+                let state = state.clone();
+                let op = op.clone();
+                let morsel = morsel.clone();
+                let span = span.clone();
+                let rt_context = rt_context.clone();
+                let fut = async move {
+                    let mut state_guard = state.lock().await;
```

Comment:
don't love that we have to lock outside the loop, especially if the op has a default state. I think it would be better to pass in some kind of `State` struct that has the `Mutex<Box<dyn OpState>>` inside of it. Then the op has the choice to lock if needed. The main gotcha I see with that however is that the function then would have to be async if we continue to use tokio mutex

Reply:
Made a `StateWrapper` struct that holds a `std::sync::Mutex<Box<dyn OpState>>`